### PR TITLE
Use locator for exisiting raw files

### LINKED ISSUE #388: Use resource locator for raw textures to find previouly used raw textures
The following raw textures are loaded differently than misc0a.raw
* Sky.raw
* smallbumpa.raw

They should be available as the following
* raw/Sky
* raw/smallbumpa

### PR DESCRIPTION
**Pull request by bwrsandman**
Also added Skya to the water shader (diffuse is confusingly called Sky for water) for alpha but with alpha blending off, it has no effect 
Resolves #388

### CODE REVIEW

File: src/Renderer.cpp [hunk 1]
```diff
@@ -443,9 +448,12 @@ void Renderer::DrawPass(const DrawSceneDesc& desc) const
 		{
 			for (auto& block : desc.island.GetBlocks())
 			{
+				static const auto textureId = entt::hashed_string("raw/smallbumpa");
```
Comment (bwrsandman) [inline src/Renderer.cpp:451]: This constant should be on the class

File: src/Renderer.cpp [hunk 2]
```diff
@@ -428,8 +428,13 @@ void Renderer::DrawPass(const DrawSceneDesc& desc) const
 			mesh->GetIndexBuffer().Bind(mesh->GetIndexBuffer().GetCount(), 0);
 			mesh->GetVertexBuffer().Bind();
 			bgfx::setState(BGFX_STATE_DEFAULT);
-			waterShader->SetTextureSampler("s_diffuse", 0, *desc.water._texture);
-			waterShader->SetTextureSampler("s_reflection", 1, desc.water.GetFrameBuffer().GetColorAttachment());
+			static const auto diffuseId = entt::hashed_string("raw/Sky");
+			static const auto alphaId = entt::hashed_string("raw/Skya");
```
Comment (bwrsandman) [inline src/Renderer.cpp:432]: These constants should be on the class

File: src/3D/LandIsland.cpp [hunk 3]
```diff
@@ -31,14 +30,6 @@ const float LandIsland::CellSize = 10.0f;
 LandIsland::LandIsland()
     : _blockIndexLookup {0}
 {
-	auto& filesystem = Game::instance()->GetFileSystem();
-	auto file = filesystem.Open(filesystem.TexturePath() / "smallbumpa.raw", FileMode::Read);
-	auto* smallbumpa = new uint8_t[file->Size()];
-	file->Read(smallbumpa, file->Size());
-
-	_textureSmallBump = std::make_unique<Texture2D>("LandIslandSmallBump");
-	_textureSmallBump->Create(256, 256, 1, Format::R8, Wrapping::Repeat, smallbumpa, static_cast<uint32_t>(file->Size()));
```
Comment (raffclar) [inline src/3D/LandIsland.cpp:40]: Getting odd visual artifacts on my end. Different format perhaps?

image

Comment (raffclar) [reply src/3D/LandIsland.cpp:40]: Logic within the loader for comparison:

```texture->Create(width, height, 1, format, graphics::Wrapping::ClampEdge, data.data(), static_cast<uint32_t>(data.size()));```

File: src/3D/Water.cpp [hunk 4]
```diff
@@ -28,16 +28,6 @@ Water::Water()
...
 	    std::make_unique<FrameBuffer>("Reflection", static_cast<uint16_t>(1024), static_cast<uint16_t>(1024),
 	                                  graphics::Format::RGBA8, graphics::Format::Depth24Stencil8);
-
-	// water texture (256x256 RAW RGB 24bpp)
-	auto& filesystem = Game::instance()->GetFileSystem();
-	auto const& waterTextureData = filesystem.ReadAll(filesystem.TexturePath() / "Sky.raw");
-	const uint16_t textureWidth = 256, textureHeight = 256;
-
-	_texture = std::make_unique<Texture2D>("Water");
-	_texture->Create(textureWidth, textureHeight, 1, Format::RGB8, Wrapping::Repeat, waterTextureData.data(),
```
Comment (raffclar) [inline src/3D/Water.cpp:38]: Same issue here. The water plane has no texture and the above differs from the texture loader

image

**Review by raffclar — CHANGES_REQUESTED** [review]

**Review by raffclar — APPROVED** [review]
Nice one 👍

**Pull request merged**